devito/finite_difference.py

```diff
@@ -221,3 +221,62 @@ def second_cross_derivative(function, dims, order):
     """
     first = second_derivative(function, dim=dims[0], width=order)
     return second_derivative(first, dim=dims[1], order=order)
+
+
+def staggered_diff(f, dim, order, stagger=centered, theta=0, phi=0):
+    """
+    Utility function to generate staggered derivatives
+    :param f: function objects, eg. `f(x, y)` or `g(t, x, y, z)`.
+    :param dims: symbol defining the dimension wrt. which
+       to differentiate, eg. `x`, `y`, `z` or `t`.
+    :param order: Order of the coefficient discretization and thus
+                  the width of the resulting stencil expression.
+    :param stagger: Shift for the FD, `left`, `right` or `centered`
+    :param theta: Dip (or polar) angle for rotated FD
```

Comment:
I agree these typically tti-specific names could be changed

Reply:
Added polar (conventional name) and azimuth already conventional name

Reply:
but the variable names stayed the same?

Reply:
also, do/should we support ``polar/azimuth`` in generic_derivative too?

Reply:
>but the variable names stayed the same?
yeah still theta/phi is convention.

>do/should we support polar/azimuth in generic_derivative too?

Maybe, currently only inside `tti/operator` as only placed used and a bit weirdly done to improve tti. Out of scope of this PR I hink

Reply:
agreed this can be done in another PR